src/main/java/com/example/gildedrose/service/ItemUpdater.java

```diff
@@ -17,6 +17,7 @@ public ItemUpdater(ItemRepository itemRepository) {
         this.itemRepository = itemRepository;
     }
 
+    @Scheduled(cron = "*/5 * * * * ?")
```

Comment:
**question** Is there a way that we could make this configurable?

It feels like `application.properties` or `application.yml` might be a good place to add this add configuration.

Reply:
Also, I might be inclined to create a dedicated component for my scheduled tasks. The schedule feels more like configuration than part of the service definition.

```java
@Component
class ScheduledTasks {
    ScheduledTasks(ItemUpdater updater) {
        this.updater = updater
    }

    @Scheduled("...")
    public updateItems() {
        // Maybe log that items are about to be updated? 
        this.updater.updateItems()
       // Do something after like logging?
    }
}
```

It might be a little redundant if the scheduled method only delegates to the service method, but this would be a convenient place to put schedule related logging. We might not want to litter our service with those details.

Additionally, this separates the scheduling responsibility and the item updating responsibility.

Reply:
Great idea, thanks for the suggestion